### PR TITLE
mon, mgr: Make 'pgs_brief' as default

### PR DESCRIPTION
**Pull request by pponnuvel**
When doing PG dump using 'ceph pg dump --format json-pretty' the output is extremely big that the command hangs and also the ceph-mgr hangs and eventuall fails over.

The exact size depends on the number of OSDs in the cluster and the number of peers for each OSD.

Fixes https://tracker.ceph.com/issues/57460

Signed-off-by: Ponnuvel Palaniyappan <pponnuvel@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by pponnuvel** [comment]
jenkins test make check arm64

**Review by ljflores — COMMENTED** [review]
@pponnuvel I read through the initial tracker. @rzarzynski mentioned using a feature flag to differentiate the behavior. Are there plans to add that to this implementation?

**Comment by pponnuvel** [comment]
jenkins test api

**Comment by pponnuvel** [comment]
jenkins test api

**Comment by pponnuvel** [comment]
jenkins test api

**Comment by pponnuvel** [comment]
@rzarzynski @ljflores I've feature-gated this change.
I've also updated the release notes. I am unsure if this is the right place or even this change warrants.  Please review and let me know if this is OK. Thanks.

File: src/mon/PGMap.cc [hunk 1]
```diff
@@ -3513,8 +3513,13 @@ int process_pg_map_command(
       copy(dumpcontents.begin(), dumpcontents.end(),
            inserter(what, what.end()));
     }
-    if (what.empty())
-      what.insert("all");
+    if (what.empty()) {
+      if (CEPH_FEATURE_SERVER_REEF) {
```
Comment (rzarzynski) [inline src/mon/PGMap.cc:3517]: The reef release is quite soon while the patch changes the output of a pretty frequently used command. This makes me worried about people who scripted it.

Though the patch looks technically fine, I would recommend bring it to next CDM (as it's an interface change) and merging after Reef is out.

Comment (pponnuvel) [reply src/mon/PGMap.cc:3517]: Thanks @rzarzynski. I can bring this up at the next CDM!

Comment (pponnuvel) [reply src/mon/PGMap.cc:3517]: @rzarzynski I realize that today's CDM is at 21:00 EDT which is 2:00 AM my local time! Is there anything else I could do instead if I can't attend it?

**Comment by pponnuvel** [comment]
@rzarzynski Hi!
Just checking if this could be looked at now as Reef seems to be released <sup>1</sup> now. Thanks!

1. https://www.spinics.net/lists/dev-ceph/msg05348.html

**Comment by rzarzynski** [comment]
@pponnuvel: Hello! First of all, sorry for the late reply.

Yes, now Reef is out, so let's get back to this stuff. As always with operator-visible changes, a discussion is needed. I'm adding a label for that and advising to bring the topic to a CDM (https://tracker.ceph.com/projects/ceph/wiki/Planning).

CCing @ljflores for planning CDM sessions in advance.

**Comment by pponnuvel** [comment]
No worries @rzarzynski ! I was well aware of the reasons Reef was delayed and thus forcing this to wait :)

Btw, I have just rebased this with the latest main as this behind. And looks fine.

I was actually intending to attend the Sept CDM if it's in my TZ (GMT+1). Otherwise I believe there's sufficient info in the tracker to explain the background. Please let me know if you need anything else from me. Thanks again!

**Comment by pponnuvel** [comment]
jenkins test make check arm64

**Comment by pponnuvel** [comment]
jenkins test make check

**Comment by pponnuvel** [comment]
jenkins test make check arm64

**Comment by pponnuvel** [comment]
jenkins test make check arm64

**Comment by pponnuvel** [comment]
jenkins test make check arm64

**Comment by pponnuvel** [comment]
jenkins test make check arm64

**Comment by ronen-fr** [comment]
Note that this breaks multiple QA scripts. Fixing those scripts should be part of the PR.

**Comment by pponnuvel** [comment]
jenkins test make check

**Pull request closed without merging**